### PR TITLE
nixos/jellyfin: Add dataDir and cacheDir to module

### PR DESCRIPTION
**Pull request by peperunas**
###### Description of changes

Added two new options for Jellyfin's module: `dataDir` and `cacheDir`.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/jellyfin.nix [hunk 1]
```diff
@@ -52,9 +64,9 @@ in
       serviceConfig = rec {
         User = cfg.user;
         Group = cfg.group;
-        StateDirectory = "jellyfin";
-        CacheDirectory = "jellyfin";
-        ExecStart = "${cfg.package}/bin/jellyfin --datadir '/var/lib/${StateDirectory}' --cachedir '/var/cache/${CacheDirectory}'";
+        StateDirectory = cfg.dataDir;
+        CacheDirectory = cfg.cacheDir;
```
Comment (pennae) [inline nixos/modules/services/misc/jellyfin.nix:68]: these are invalid according to systemd.exec(5):
```
       RuntimeDirectory=, StateDirectory=, CacheDirectory=, LogsDirectory=, ConfigurationDirectory=
           These options take a whitespace-separated list of directory names. The specified directory
           names must be relative, and may not include "..".
```

Comment (pennae) [reply nixos/modules/services/misc/jellyfin.nix]: don't know enough about jellyfin to make an informed decision, but it feels like we should bind mount them instead if possible. systemd will create these directories if they don't exist, that's a good bit of convenience for the default setup and something we'd have to emulate in a preStart script if they went away.

Comment (peperunas) [reply nixos/modules/services/misc/jellyfin.nix]: How do you suggest me to proceed in this case then @pennae?

Comment (pennae) [reply nixos/modules/services/misc/jellyfin.nix]: easiest way would be to hardcode both to `jellyfin` and have users who want to use different directories bind mount those by adding their own `BindPaths`. most users *probably*(?) won't want to change those anyway.

**Pull request closed without merging**